engine/table/src/main/java/io/deephaven/stream/StreamToBlinkTableAdapter.java

```diff
@@ -277,50 +212,49 @@ public Table table() {
 
     @Override
     public void close() {
-        if (!alive) {
-            return;
-        }
-        synchronized (this) {
-            if (!alive) {
-                return;
-            }
-            alive = false;
+        if (alive.compareAndSet(true, false)) {
             log.info().append("Deregistering ").append(StreamToBlinkTableAdapter.class.getSimpleName()).append('-')
                     .append(name)
                     .endl();
             updateSourceRegistrar.removeSource(this);
-            final Runnable localShutdownCallback = shutdownCallback;
-            if (localShutdownCallback != null) {
-                localShutdownCallback.run();
-            }
+            streamPublisher.shutdown();
         }
     }
 
-    @Override
-    public void destroy() {
-        close();
-    }
-
     @Override
     public void run() {
+        synchronized (this) {
+            // If we have an enqueued failure we want to process it first, before we allow the streamPublisher to flush
+            // itself.
+            if (enqueuedFailures != null) {
+                deliverFailure(MultiException.maybeWrapInMultiException(
```

Comment:
Do we need to concern ourselves w/ the potential for "double delivery" of these errors? I'm not sure the exact semantics in these layers, but we could easily guard against that if necessary.

Reply:
No, because we close immediately afterwards, ensuring that our `run()` method will not be called again. Otherwise, it would be a concern.